geoips/commandline/ancillary_info.py

```diff
@@ -0,0 +1,95 @@
+"""Module for retrieving help information for the GeoIPS Command Line Interface.
+
+Used to dynamically load and apply help information to all 'geoips' commands.
+"""
+
+import json
+from os.path import dirname, exists, getmtime
+import yaml
+
+
+def cmd_instructions_modified(ancillary_dirname):
+    """Check whether or not cmd_instructions.yaml has been modified.
+
+    This uses os.path.getmtime(fname) determine whether or not the YAML command help
+    instructions have been modified more recently than when we last generated our JSON
+    instructions file. Return the truth value to whether or not cmd_instructions.yaml
+    has been modified more recently than cmd_instructions.json
+
+    Parameters
+    ----------
+    ancillary_dirname: str
+        - The path to the folder which contains the help instructions for the CLI
+
+    Returns
+    -------
+    yaml_recently_modified: bool
+        - The truth value as to whether or not the yaml cmd_instructions were modified
+          more recently than the json cmd_instructions
+    """
+    json_mtime = getmtime(f"{ancillary_dirname}/cmd_instructions.json")
+    yaml_mtime = getmtime(f"{ancillary_dirname}/cmd_instructions.yaml")
+    yaml_recently_modified = False
+    if yaml_mtime > json_mtime:
+        # yaml file was modified more recently than json_mtime
+        yaml_recently_modified = True
+    return yaml_recently_modified
+
+
+def get_cmd_instructions(ancillary_dirname=None):
+    """Return a dictionary of instructions for each command, obtained by a yaml file.
+
+    This has been placed as a module attribute so we don't perform this process for
+    every CLI sub-command. It was taking too long to initialize the CLI and this was a
+    large partof that. See
+    https://github.com/NRLMMD-GEOIPS/geoips/pull/444#discussion_r1541864672 for more
+    information.
+
+    For more information on what's available, see:
+        geoips/commandline/ancillary_info/cmd_instructions.yaml
+
+    Parameters
+    ----------
+    ancillary_dirname: str
+        - The path to the folder which contains the help instructions for the CLI.
+          Defaults to None in case a user wants to supply a different path for testing
+          purposes
+
+    Returns
+    -------
+    cmd_instructions: dict
+        - Dictionary of help instructions for every CLI sub-command
+    """
```

Comment:
How long does this take each time the CLI is called? We can figure this out using `cprofile`. If this is slow, we might consider creating a new issue to split this up across subcommands or something like that.

Maybe just create an issue to assess the speed of this command and mitigate if it is slow.